### ISSUE TITLE
androidTest failure on Android 13 / API33 because of notification permission prompt

### ISSUE THREAD

**Issue by mikehardy** [ankidroid/Anki-Android#12194]
###### Reproduction Steps

1. make a new emulator and suffer through the download of the API33 image
2. fire up that emulator
3. `./gradlew jacocoAndroidTestReport`

###### Expected Result

Everything is awesome

###### Actual Result

You can see the notification prompt popup for the app as it starts on the emulator while the tests are running, and one of them fails

We probably need to implement the new Android 13 permission request for notifications

**Comment by n1snt** [comment]
What I think needs to be done:

1. Bump the compileSdkVersion to 33 (Tried bumping the compileSdkVersion to 33 and it threw a lot of errors compiling the app)
2. Add the new notification permission to `AndroidManfests.xml`
`<uses-permission android:name="android.permission.POST_NOTIFICATIONS"/>`
 _[Source](https://developer.android.com/develop/ui/views/notifications/notification-permission)_
3. Add `Manifest.permission.POST_NOTIFICATIONS` to `mRuntimePermissionRule` in `DeckPickerTest`.
Something like this: (Just an approx idea of what needs to be added.)
```
    @get:Rule
    val mRuntimePermissionRule: GrantPermissionRule =
        if (Build.VERSION.SDK_INT >= 33) {
            GrantPermissionRule.grant(Manifest.permission.WRITE_EXTERNAL_STORAGE,
                Manifest.permission.POST_NOTIFICATIONS)
        } else {
            GrantPermissionRule.grant(Manifest.permission.WRITE_EXTERNAL_STORAGE)
        }
```

**Comment by mikehardy** [comment]
That does seem about right

**Comment by n1snt** [comment]
The test seems to be passing now.
It looks like we have still not added the notification permission. [Source](https://developer.android.com/develop/ui/views/notifications/notification-permission)

**Comment by mikehardy** [comment]
The test passes for me after I accept permission on the emulator, it appears to be persistent. It's the fresh install on fresh emulator case that was struggling 

compile version is 33 now from my David's work on #12359 

Looks like I actually fixed this in #12643 but didn't close it :laughing:
